vnets.tf

```diff
@@ -58,14 +55,33 @@ resource "azurerm_virtual_network" "private" {
   tags                = local.default_tags
 }
 
-# Dedicated subnet for external access
+# Dedicated subnet for external access (such as VPN external NIC)
 resource "azurerm_subnet" "dmz" {
   name                 = "${azurerm_virtual_network.private.name}-dmz"
   resource_group_name  = azurerm_resource_group.private.name
   virtual_network_name = azurerm_virtual_network.private.name
   address_prefixes     = ["10.248.0.0/28"]
 }
 
+# Dedicated subnet for machine to machine private communications
+resource "azurerm_subnet" "data_tier" {
```

Comment:
```suggestion
resource "azurerm_subnet" "private_vnet_data_tier" {
```

Reply:
This change requires the command `terraform state mv 'azurerm_subnet.data_tier' 'azurerm_subnet.private_vnet_data_tier'` to be run. 

I'm editing the PR message + restart a build to check the latest plan.